README.md

```diff
@@ -4,110 +4,167 @@
+provides a consistent interface to interact with OpenWhisk services. With this
+tool to download and configure, you are able to manage OpenWhisk services from
+the command line and automate them through scripts.
 
-The OpenWhisk Command Line Interface (OpenWhisk CLI) is a unified tool that provides a consistent interface to
-interact with OpenWhisk services. With this tool to download and configure, you are able to manage OpenWhisk services
-from the command line and automate them through scripts.
+# Where to download the binary of OpenWhisk CLI
 
+The OpenWhisk CLI is available on the release page: [click here to
+download](https://github.com/apache/incubator-openwhisk-cli/releases). We
+currently have binaries available for Linux, Mac OS and windows under i386 and
+amd64 architectures.  Linux versions are also available under Linux on Z, Power
+and 64-bit ARM architectures.  You can download the binary, which fits your
+local environment.
 
-# Where to download the binary of OpenWhisk CLI
+# How to build the binary locally
 
-The OpenWhisk CLI is available on the release page: [click here to download](https://github.com/apache/incubator-openwhisk-cli/releases).
-We currently have binaries available for Linux, Mac OS and windows under amd64 architecture. You can download the
-binary, which fits your local environment.
+The OpenWhisk CLI is written in the Go language. You have two options to build
+the binary locally:
 
+1.  Build using the packaged Gradle scripts (including the 'gogradle' plugin),
+now the preferred build method.
+2.  Compile in your local Go environment,
 
-# How to build the binary locally
+## Build the binary with Gradle
 
-You can also choose to build the binaries locally from the source code with Go tool.
+**Note:** For those who may have used the Gradle build previously, it has been
+re-engineered to no longer required Docker or Go to be pre-installed on your
+system.  Using the [gogradle](https://github.com/gogradle/gogradle) plugin,
+Gradle now uses a prexisting Go environment to build if it can be located, or
+downloads and installs an environment within the build directory.
 
-Make sure that you have Go installed [installing Go](https://golang.org/doc/install), and `$GOPATH` is defined [Go development environment](https://golang.org/doc/code.html).
+To build with Gradle, open an terminal, go to the directory of OpenWhisk CLI
+home directory, and build the binary via the following command under Linux or
+Mac:
 
-Then download the source code of the OpenWhisk CLI and the dependencies by typing:
+```
+$ ./gradlew compile -PnativeCompile
+```
+
+or run the following command for Windows:
 
 ```
-$ cd $GOPATH
-$ go get github.com/apache/incubator-openwhisk-cli
+$ ./gradlew.bat compile -PnativeCompile
 ```
 
+After the build, you can find the binary `wsk` or `wsk.exe` in the bin folder
+under the OpenWhisk CLI home directory. In addition, it is also available under
```

Comment:
Need to remove these two sentences that reference the `bin` folder.

Reply:
Re-written.